rust/src/main.rs

```diff
@@ -75,12 +84,26 @@ async fn main() -> Result<(), Box<dyn Error>> {
         return Ok(());
     }
 
-    if let Some(name) = args.lookup.as_deref() {
-        print!("{}", client.lookup(name).await.unwrap());
+    if let Some(key) = args.lookup.as_deref() {
+        print!("{}", client.lookup(key).await.unwrap());
         return Ok(());
     }
 
-    Ok(())
+    match &args.command {
+        Some(Commands::List {}) => {
+            list_all(&client).await;
+            Ok(())
+        }
+        Some(Commands::Load { key }) => {
+            print!("{}", client.lookup(key).await.unwrap());
+            Ok(())
+        }
+        Some(Commands::Store { key, value }) => {
```

Comment:
Here we need to implement the -w flag with an `exists` library implementation for this to be viable, as we want to have a sanity check for the user to know if he/she is overwriting something.